main_window/competition/mapping/mapping_thread.py

```diff
@@ -185,26 +215,31 @@ def run(self) -> None:
                     time.sleep(0.5)
 
                 # copy location values to use, to keep the values consistent in synchronous but adjacent calls
-                latitude = self.rocket_data.last_value_by_device(self.device, DataEntryIds.LATITUDE)
-                longitude = self.rocket_data.last_value_by_device(self.device, DataEntryIds.LONGITUDE)
+                latitudes = [None]*len(self.viewed_devices)
+                longitudes = [None]*len(self.viewed_devices)
+
+                for i, device in enumerate(self.viewed_devices):
+                    latitudes[i] = self.rocket_data.last_value_by_device(device, DataEntryIds.LATITUDE)
+                    longitudes[i] = self.rocket_data.last_value_by_device(device, DataEntryIds.LONGITUDE)
+
```

Comment:
```suggestion
                latitudes = dict()
                longitudes = dict()

                for device in self.viewed_devices:
                    latitudes[device] = self.rocket_data.last_value_by_device(device, DataEntryIds.LATITUDE)
                    longitudes[device] = self.rocket_data.last_value_by_device(device, DataEntryIds.LONGITUDE)

```

I think if we use a dict here, we can get rid of the indexes here as well making the logic simpler.